On The Map/OTMClient.swift

```diff
@@ -0,0 +1,192 @@
+    var uniqueKey: String? = nil
+    var currentStudent: StudentInformation? = nil
+    var students: [StudentInformation]? = nil
+    
+    // Store the data in a central spot
+    var studentInformation: [StudentInformation] = [StudentInformation]()
+    
+    // MARK: Initializers
+    
+    override init() {
+        super.init()
+    }
+    
+    // MARK: - POST
+    
+    func taskForPOSTMethod(method: String, udacity: Bool, parameters: [String:AnyObject]?, jsonBody: [String:AnyObject], completionHandlerForPOST: (result: AnyObject!, error: NSError?) -> Void) -> NSURLSessionDataTask {
+        
+        /* 1. Set the parameters */
+        var urlString: String
+        
+        if let methodParameters = parameters {
+            urlString = method + OTMClient.escapedParameters(methodParameters)
+        } else {
+            urlString = method
+        }
+        
+        /* 2/3. Build the URL, Configure the request */
+        let url = NSURL(string: urlString)!
+        let request = NSMutableURLRequest(URL: url)
+        request.HTTPMethod = "POST"
+        
+        /* Is the request Udacity or Parse? */
+        if udacity {
+            request.addValue("application/json", forHTTPHeaderField: "Accept")
+            request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+            
+        } else {
+            request.addValue("QrX47CA9cyuGewLdsL7o5Eb8iug6Em8ye0dnAbIr", forHTTPHeaderField: "X-Parse-Application-Id")
+            request.addValue("QuWThTdiRmTux3YaDseUSEpUKo7aBYM737yKd4gY", forHTTPHeaderField: "X-Parse-REST-API-Key")
+        }
+        
+        do {
+            request.HTTPBody = try NSJSONSerialization.dataWithJSONObject(jsonBody, options: [])
+        } catch _ as NSError {
+            request.HTTPBody = nil
+        }
+        
+    
+        /* 4. Make the request */
+        let task = session.dataTaskWithRequest(request) { (data, response, error) in
+            
+            if let JSONError = error {
+                
+                _ = OTMClient.errorForData(data, response: response, error: JSONError)
+                print(error)
+                completionHandlerForPOST(result: nil, error: error)
+            } else {
+                var newData = data
+                if(udacity){
+                    newData = data!.subdataWithRange(NSMakeRange(5, data!.length - 5)) 
+                }
+                /* 5/6. Parse the data and use the data (happens in completion handler) */
+                self.convertDataWithCompletionHandler(newData!, completionHandlerForConvertData: completionHandlerForPOST)
+            }
+            
+        }
+        
+        /* 7. Start the request */
+        task.resume()
+        
+        return task
+    }
+    
+    // MARK: - GET
+    
+    func taskForGETMethod(method: String, udacity: Bool, parameters: [String:AnyObject]?, completionHandlerForGET: (result: AnyObject!, error: NSError?) -> Void) -> NSURLSessionDataTask {
```

Comment:
Since this class is specifically an OTMClient, I would strongly consider returning StudentInformation objects here.